sync_diff_inspector/source/source.go

```diff
@@ -134,13 +139,48 @@ func NewSources(ctx context.Context, cfg *config.Config) (downstream Source, ups
 		// When the router set case-sensitive false,
 		// that add rule match itself will make table case unsensitive.
 		for _, d := range cfg.Task.SourceInstances {
-			if d.Router.AddRule(&router.TableRule{
-				SchemaPattern: tableConfig.Schema,
-				TablePattern:  tableConfig.Table,
-				TargetSchema:  tableConfig.Schema,
-				TargetTable:   tableConfig.Table,
-			}) != nil {
-				return nil, nil, errors.Errorf("set case unsensitive failed. The schema/table name cannot be parttern. [schema = %s] [table = %s]", tableConfig.Schema, tableConfig.Table)
+			if _, ok := d.RouteTargetSet[dbutil.TableName(tableConfig.Schema, tableConfig.Table)]; ok {
+				// There is a user rule routing to `tableConfig.Schema`.`tableConfig.Table`
```

Comment:
Maybe we can add a case in comment here to describe the behavior here:

| sourceDB | targetDB |
| :---: | :---: |
| test1 | test |
| test2 | test2 |         
| test | not int routes |

In this case we will check test1 -> test, test2 -> test2. But we won't check schema test.

Reply:
sure, add unit test case in https://github.com/pingcap/tidb-tools/pull/611/commits/267c3d6bc7e79ca41d1936bb1793b58907bea109. PTAL